### ISSUE TITLE
Kurseditor Autotool-Aufgabe: Änderung einer laufenden Aufgabe ist unmöglich

### ISSUE THREAD

**Issue by jwaldmann** [klemens/openolat#63]
Anwendungsfall: Aufgabe wurde gestellt. Nach Beginn Bewertungszeitraum wird klar,
daß Konfiguration geändert werden sollte. (Bsp.: While-Programme: die Obergrenze für 
die Schrittzahl im Simulator soll erhöht werden)

Ist-Zustand: das geht gar nicht (man kann die Konf. nicht editieren).

Soll: das soll eben doch irgendwie gehen. Bereits erreichte Bewertungspunkte sollten dabei
nicht verschwinden. (Im Anwendungsfall konnte aber noch niemand Punkte erreicht haben,
eben weil der Simulator niemals weit genug rechnete.)

Work-Around: alte Aufgabe löschen, neue einfügen. (Dabei würden aber Punkte
verlorengehen.)

**Comment by klemens** [comment]
Sobald ein Student die Aufgabe öffnet, wird für sie/ihn eine persönliche Konfiguration angelegt und ab diesem Zeitpunkt kann die Aufgabe nicht mehr geändert werden.

> Bereits erreichte Bewertungspunkte sollten dabei nicht verschwinden. 

Gerade aus diesem Grund wird eben die Änderung der Aufgabe verboten: Um zu verhindern, dass einige Studenten einen Vor- oder Nachteil haben.
Zum Testen der Aufgabenkonfiguration vor der Veröffentlichung kann ja der Bereich "Aufgabenkonfiguration testen" bei der Aufgabe verwendet werden.

Ein weiteres Problem besteht darin, dass bei einer geänderten Aufgabe eine vor der Änderung gespeicherte, korrekte Lösung möglicherweise nicht mehr korrekt ist. (Außer diese Studenten behalten dann die alte Aufgabenkonfiguration..)

**Comment by jwaldmann** [comment]
ja, das ist eine Designfrage. Ich denke: ob ein es ungerechtfertigter Vorteil ist, daß bereits erzielte Punkte bestehen bleiben, sollte doch der Aufgabesteller im Einzelfall selbst entscheiden können. 

"Zum Testen vor der Veröffentlichung ..." jaja, aber nicht jeder Fehler wird sofort gefunden.
